src/common/cluster/authorization-namespace-review.injectable.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type { KubeConfig } from "@kubernetes/client-node";
+import { AuthorizationV1Api } from "@kubernetes/client-node";
+import logger from "../logger";
+import { getInjectable } from "@ogre-tools/injectable";
+
+export type NamespaceResources = (namespace: string) => Promise<string[]>;
+
+/**
+ * @param proxyConfig This config's `currentContext` field must be set, and will be used as the target cluster
+   */
+export function authorizationNamespaceReview(proxyConfig: KubeConfig): NamespaceResources {
```

Comment:
1. This really shouldn't be exported
2. This should probably be a builder of sorts which first takes `dependencies: Dependencies` and then returns this current function. That way you can inject `logger` in.

```ts
export type RequestNamespaceResources = (namespace: string) => Promise<string[]>;
export type AuthorizationNamespaceReview = (proxyConfig: KubeConfig) => RequestNamespaceResources;

interface Dependencies {
  logger: Logger;
}

const authorizationNamespaceReview( = ({
  logger,
}: Dependencies): AuthorizationNamespaceReview => {
  ...
};
```